mlir/lib/Transforms/RemoveDeadValues.cpp

```diff
@@ -598,6 +598,9 @@ void RemoveDeadValues::runOnOperation() {
       // Nothing to do because this terminator is associated with either a
       // function op or a region branch op and gets cleaned when these ops are
       // cleaned.
+    } else if (op->hasTrait<::mlir::OpTrait::IsTerminator>()) {
+      // Nothing to do here because this a terminator op and it should be
+      // honored with respect to its parent
```

Comment:
Seems in line with the `cleanSimpleOp` comment that states:
```
/// It is assumed that `op` is simple. Here, a simple op is one which isn't a
/// symbol op, a symbol-user op, a region branch op, a branch op, a region
/// branch terminator op, or return-like.
```

Now isn't this check redundant with `ReturnLike` and `RegionBranchTerminatorOpInterface`? There two should also be Terminator.

Reply:
The check seems completely relevant for gpu.termnator op as it [declared](https://github.com/llvm/llvm-project/blob/72ce6294157964042b7ed5576ce2c99257eeea9d/mlir/include/mlir/Dialect/GPU/IR/GPUOps.td#L952) as:

>  952 def GPU_TerminatorOp : GPU_Op<"terminator", [HasParent<"LaunchOp">,
>  953                                              Pure, Terminator]>,
>  954     Arguments<(ins)>, Results<(outs)> {
> 
Same for gpu.return op

Reply:
Which check are you referring to? The new one?

You may have read it the opposite of what I wrote: I am referring to the **existing** checks being obsolete after this patch.

Reply:
Sorry, misread you.

Reply:
OK, so: can we remove them?

Reply:
Removed.